File: 2021/09/23/main.py
class Solution:
    def breakPalindrome(self, palindrome: str) -> str:
        i, n = 0, len(palindrome)
        if n == 1:
            return ''
        l = list(palindrome)
        while i < n:
            if l[i] != 'a' and i != n-i-1:
                l[i] = 'a'
                break
            i += 1
        if i == n:
            l[i-1] = chr(ord(l[i-1]) + 1)
        
        return ''.join(l)

File: 2021/09/23/main.go
package main

func breakPalindrome(palindrome string) string {
	i, n := 0, len(palindrome)
	if n == 1 {
		return ""
	}
	l := []byte(palindrome)
	for i < n {
		if l[i] != 'a' && i != n-i-1 {
			l[i] = 'a'
			break
		}
		i += 1
	}
	if i == n {
		l[i-1] = byte(int(l[i-1]) + 1 + 'a')
	}

	return string(l)
}
